internal/apex/ctlserver.go

```diff
@@ -0,0 +1,84 @@
+package apex
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+	"net/rpc/jsonrpc"
+	"os"
+)
+
+// TODO make this path configurable
+const UnixSocketPath = "/var/run/apex.sock"
+
+// Assumptions made in here:
+//  - ax.logger is thread-safe
+
+func (ax *Apex) CtlServerStart() error {
+	switch ax.os {
+	case Linux.String():
+		return ax.CtlServerStartLinux()
+
+	case Darwin.String():
+		ax.logger.Info("Ctl interface not yet supported on OSX")
+
+	case Windows.String():
+		ax.logger.Info("Ctl interface not yet supported on Windows")
+	}
+
+	return nil
+}
+
+type ApexCtl struct {
+	ax *Apex
+}
+
+func (ac *ApexCtl) Status(_ string, result *string) error {
```

Comment:
@russellb we also need a `func (ac *ApexCtl) Version() {}` RPC since there is a possibility of client-server drift.
i.e we `make dist/apex` but forget to `make dist/apexd`

The version information should be added at compile time with ldflags https://www.digitalocean.com/community/tutorials/using-ldflags-to-set-version-information-for-go-applications

We can roll with just making sure that the client/daemon are the exact same version for now.

Reply:
Good call. I'll add that. I can also make a `version` subcommand to `apex` while I'm at it.